contracts/Fund.sol

```diff
@@ -65,11 +68,10 @@ contract Fund is Administratable {
   /**
    * @notice Change Fund Primary Advisor
    * @param  newManager The address of the new PrimaryAdvisor.
-   * @param  adminContractAddress Address of the EndaomentAdmin contract.
    */
-  function changeManager(address newManager, address adminContractAddress)
+  function changeManager(address newManager)
     public
-    onlyAdminOrRole(adminContractAddress, IEndaomentAdmin.Role.REVIEWER)
+    onlyAdminOrRole(fundFactoryContract.endaomentAdmin(), IEndaomentAdmin.Role.REVIEWER)
```

Comment:
Wouldn't it be simpler to amortize the call to fundFactoryContract.endaomentAdmin by just storing the endaomentAdmin as a state var and not the whole fund factor? Similarly else where?

Reply:
Similarly, couldn't we then remove the need to pass the modifiers the endaomentAdmin entirely by making them state variables of the Administratable contract?

Reply:
> Wouldn't it be simpler to amortize the call to fundFactoryContract.endaomentAdmin by just storing the endaomentAdmin as a state var and not the whole fund factor? Similarly else where?

The issue with this is that the admin is no longer "reactive". In other words, if you update the admin in `FundFactory`, the individual funds do not know it's updated if you hardcode it in `Fund. By querying each time like we currently do, we ensure they always are using the correct admin contract

> Similarly, couldn't we then remove the need to pass the modifiers the endaomentAdmin entirely by making them state variables of the Administratable contract?

Hm, I'm not sure I'm following. Can you clarify?

Reply:
> Similarly, couldn't we then remove the need to pass the modifiers the endaomentAdmin entirely by making them state variables of the Administratable contract?

Ah, I see. I think this is what they referred to in the audit report. The downside here is that this requires `Administratable` to have a constructor, where `endaomentAdmin` is set. If `FundFactory`, `Fund`, etc. all inherit from `Administrable`, they each deploy their own instance and set `endaomentAdmin` in the constructor. This means that (1) you deploy lots of `Administratable` contracts, and (2) If you want to update the admin, you have to do it in each contract individually, which is not practical

cc @apbendi to double check me here

Reply:
Correct. We want to reuse the `Administratable` functionality (i.e. the modifiers) on both `(Fund|Org)Factory` and `(Fund|Org)` *without* tying the state variable to `Administratable`, because if we do, each `(Fund|Org)` would have to be updated individually. That's why we inject the addr into the modifiers, and only store the addr on the `(Fund|Org)Factory`.

Reply:
Okay sounds good.